### PR TITLE
[Cute, Fwd] Add return_entropy to compute per-row attention entropy

### LINKED ISSUE #1878: Could FA return entropy of each row?
As we all know, FA could return LSE. Could FA return entropy of each row?

### PR DESCRIPTION
**Pull request by NJX-njx**
## Summary
- Add `return_entropy` parameter to `flash_attn_func` and `flash_attn_varlen_func` that returns per-row attention entropy `H = -sum(p_i * log(p_i))` computed online alongside the forward pass
- The entropy is computed incrementally using a `weighted_score_sum` accumulator in the online softmax loop, with zero overhead when `return_entropy=False`
- Useful for model analysis, training monitoring, and entropy-based regularization

## Math
The entropy `H = -sum(p_i * log(p_i))` can be rewritten as:

```
H = lse - E[s] = lse - (1/row_sum) * sum(exp(s_i - row_max) * s_i)
```

where `s_i = score_i * softmax_scale`. We track `weighted_score_sum = sum(exp(s_i - row_max) * s_i)` incrementally across K blocks, rescaling when `row_max` changes (same factor as `row_sum`).

## Changes
- `flash_attn/cute/softmax.py` — Add `weighted_score_sum` accumulator to `Softmax` class, accumulate in `online_softmax`, compute final entropy in `finalize`
- `flash_attn/cute/interface.py` — Parameter threading, tensor allocation, compile key, autograd functions
- `flash_attn/cute/flash_fwd.py` — Base epilogue entropy write, SM80 parameter threading
- `flash_attn/cute/flash_fwd_sm90.py` — SM90 parameter threading
- `flash_attn/cute/flash_fwd_sm100.py` — Accept parameter (not yet implemented)

## Test plan
- [ ] Verify `return_entropy=True` returns correct entropy values on SM80/SM90/SM120
- [ ] Compare against reference: `p = F.softmax(q @ k.T * scale, dim=-1); H = -(p * p.log()).sum(-1)`
- [ ] Verify `return_entropy=False` (default) has no performance impact
- [ ] Verify SM100/SM110 raises `NotImplementedError`

Closes #1878

### CODE REVIEW

**Comment by tridao** [comment]
Similar to #2357, I don't think adding these justifies the complexity and maintenance burden at the moment
